### DISCUSSION TITLE
FIWARE NGSI-LD, NGSIv2, and Ultralight 2.0 protocols

### DISCUSSION THREAD

**Discussion by amotl** [daq-tools/kotori#200] [Ideas]
## About

> Ultralight 2.0 is a lightweight text based protocol aimed to constrained devices and communications where the bandwidth and device memory may be limited resources.
>
> -- https://fiware-iotagent-ul.readthedocs.io/en/latest/usermanual.html

## Introduction
- https://en.wikipedia.org/wiki/NGSI-LD
- https://de.wikipedia.org/wiki/NGSI-LD
- https://fiware.github.io/specifications/ngsiv2/stable/
- https://fiware-tutorials.readthedocs.io/en/stable/iot-agent/

## Resources
- https://github.com/Fiware/specifications
- https://fiware-orion.readthedocs.io/en/master/devel/jsonParseV2.html
- https://fiware-orion.readthedocs.io/en/master/devel/architecture.html#a-little-bit-of-history
- https://fiware-orion.readthedocs.io/en/master/orion-api.html
- https://www.fiware.org/wp-content/uploads/2017/01/NGSIv2-Overview-for-Developers-That-Already-Know.pdf
- [ETSI GS CIM 009](https://www.etsi.org/deliver/etsi_gs/CIM/001_099/009/01.06.01_60/gs_CIM009v010601p.pdf)
- [GSMA CLP.25-v2.0](https://www.gsma.com/iot/wp-content/uploads/2018/11/CLP.25-v2.0.pdf)

**Comment by amotl** [comment]
We proposed Kotori to [awesome-europe](https://github.com/GeiserX/awesome-europe) curated by @GeiserX.

- https://github.com/GeiserX/awesome-europe/issues/18

#### TLDR;
> If Kotori can speak any NGSI variant, and if it is implemented more solidly than a thin wrapper — i.e., Kotori can genuinely participate in a FIWARE/NGSI-LD ecosystem as a data historian — then I'd be happy to consider adding it to the list. Let me know how that discussion evolves!
